core/src/main/java/org/elasticsearch/search/internal/DefaultSearchContext.java

```diff
@@ -728,6 +728,11 @@ public Counter timeEstimateCounter() {
     }
 
     @Override
+    public boolean hasFetchSubPhaseContext(FetchSubPhase.ContextFactory contextFactory) {
+        return subPhaseContexts.get(contextFactory.getName()) != null;
```

Comment:
This can just be `return subPhaseContexts.containsKey(contextFactory.getName())`